diff -ruNa a/cmake/FindOrcus.cmake b/cmake/FindOrcus.cmake
--- a/cmake/FindOrcus.cmake	2025-08-15 01:00:33.000000000 +0200
+++ b/cmake/FindOrcus.cmake	2025-11-27 23:44:19.747530767 +0100
@@ -14,21 +14,21 @@
     set (Orcus_FOUND TRUE)
 else ()
     find_package(PkgConfig QUIET)
-    pkg_search_module(PC_ORCUS liborcus liborcus-0.20 liborcus-0.19 liborcus-0.18 liborcus-0.17 liborcus-0.16 QUIET)
+    pkg_search_module(PC_ORCUS liborcus liborcus-0.21 liborcus-0.20 liborcus-0.19 liborcus-0.18 liborcus-0.17 liborcus-0.16 QUIET)
     pkg_search_module(PC_IXION libixion libixion-0.20 libixion-0.19 libixion-0.18 libixion-0.17 libixion-0.16 QUIET)
 
     find_library (Orcus_LIBRARY
-        NAMES orcus orcus-0.20 orcus-0.19 orcus-0.18 orcus-0.17 orcus-0.16
+        NAMES orcus orcus-0.21 orcus-0.20 orcus-0.19 orcus-0.18 orcus-0.17 orcus-0.16
         HINTS ${PC_ORCUS_LIBRARY_DIRS}
         PATH_SUFFIXES orcus
     )
     find_library (Orcus_parser_LIBRARY
-        NAMES orcus-parser orcus-parser-0.20 orcus-parser-0.19 orcus-parser-0.18 orcus-parser-0.17 orcus-parser-0.16
+        NAMES orcus-parser orcus-parser-0.21 orcus-parser-0.20 orcus-parser-0.19 orcus-parser-0.18 orcus-parser-0.17 orcus-parser-0.16
         HINTS ${PC_ORCUS_LIBRARY_DIRS}
         PATH_SUFFIXES orcus
     )
     find_library (Orcus_spreadsheet_LIBRARY
-        NAMES orcus-spreadsheet-model orcus-spreadsheet-model-0.20 orcus-spreadsheet-model-0.19 orcus-spreadsheet-model-0.18 orcus-spreadsheet-model-0.17 orcus-spreadsheet-model-0.16
+        NAMES orcus-spreadsheet-model orcus-spreadsheet-model-0.21 orcus-spreadsheet-model-0.20 orcus-spreadsheet-model-0.19 orcus-spreadsheet-model-0.18 orcus-spreadsheet-model-0.17 orcus-spreadsheet-model-0.16
         HINTS ${PC_ORCUS_LIBRARY_DIRS}
         PATH_SUFFIXES orcus
     )
@@ -69,7 +69,7 @@
     set(LIBIXION_VERSION ${PC_IXION_VERSION})
     # latest version
     if (NOT DEFINED LIBORCUS_VERSION)
-        set(LIBORCUS_VERSION "0.20.0")
+        set(LIBORCUS_VERSION "0.21.0")
     endif()
     if (NOT DEFINED LIBIXION_VERSION)
             set(LIBIXION_VERSION "0.20.0")
diff -ruNa a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2025-08-15 01:00:33.000000000 +0200
+++ b/CMakeLists.txt	2025-12-01 14:18:59.474239054 +0100
@@ -1,6 +1,6 @@
 # see https://community.kde.org/Policies/CMake_Coding_Style
 cmake_minimum_required(VERSION 3.17.0 FATAL_ERROR)
-project(labplot VERSION 2.12.0 LANGUAGES CXX C)
+project(labplot VERSION 2.12.1 LANGUAGES CXX C)
 set(YEAR 2025)
 
 set(CMAKE_C_STANDARD 99)
@@ -692,10 +692,11 @@
 ### Eigen (optional) #############################
 if(ENABLE_EIGEN3)
     find_package(Eigen3 QUIET)
-    if(EIGEN3_FOUND)
-	    message(STATUS "Found Eigen3 library: ${EIGEN3_INCLUDE_DIR} (version \"${EIGEN3_VERSION_STRING}\")")
+    if(Eigen3_FOUND)
+	get_target_property(EIGEN3_INCLUDE_DIR Eigen3::Eigen INTERFACE_INCLUDE_DIRECTORIES)
+	message(STATUS "Found Eigen3 library: ${EIGEN3_INCLUDE_DIR} (version \"${Eigen3_VERSION}\")")
         add_definitions(-DHAVE_EIGEN3)
-	add_definitions(-DEIGEN3_VERSION_STRING=\"${EIGEN3_VERSION_STRING}\")
+	add_definitions(-DEIGEN3_VERSION_STRING=\"${Eigen3_VERSION}\")
         include_directories(${EIGEN3_INCLUDE_DIR})
     else()
         message(STATUS "Eigen3 library NOT FOUND.")
diff -ruNa a/org.kde.labplot.appdata.xml b/org.kde.labplot.appdata.xml
--- a/org.kde.labplot.appdata.xml	2025-08-15 01:00:34.000000000 +0200
+++ b/org.kde.labplot.appdata.xml	2026-01-07 16:09:12.949454453 +0100
@@ -92,8 +92,14 @@
     <p xml:lang="uk">За допомогою LabPlot просто створювати креслення, керувати ними та редагувати креслення. За допомогою програми можна створювати креслення на основі електронної таблиці або даних, імпортованих із зовнішнього файла. Креслення можна експортувати у форматі растрового або векторного зображення.</p>
     <p xml:lang="zh-CN">LabPlot 提供了一种方便地创建、 管理和编辑图表的方式。它允许您基于电子表格的数据或从外部文件导入的数据来绘制图表。图表可以导出到多种格式的位图或矢量图形。</p>
   </description>
-  <url type="homepage">https://labplot.kde.org/</url>
+  <url type="homepage">https://labplot.org/</url>
   <url type="bugtracker">https://bugs.kde.org/enter_bug.cgi?product=LabPlot2&amp;format=guided</url>
+  <url type="faq">https://labplot.org/frequently-asked-questions/</url>
+  <url type="donation">https://labplot.org/donate/</url>
+  <url type="translate">https://l10n.kde.org/</url>
+  <url type="contact">https://labplot.org/support/</url>
+  <url type="vcs-browser">https://invent.kde.org/education/labplot/</url>
+  <url type="contribute">https://labplot.org/contribute/</url>
   <branding>
     <color type="primary" scheme_preference="light">#ffbe6f</color>
     <color type="primary" scheme_preference="dark">#c64600</color>
diff -ruNa a/src/backend/worksheet/plots/cartesian/XYFourierFilterCurve.cpp b/src/backend/worksheet/plots/cartesian/XYFourierFilterCurve.cpp
--- a/src/backend/worksheet/plots/cartesian/XYFourierFilterCurve.cpp	2025-08-15 01:00:36.000000000 +0200
+++ b/src/backend/worksheet/plots/cartesian/XYFourierFilterCurve.cpp	2025-11-27 23:36:50.625933555 +0100
@@ -25,6 +25,7 @@
 }
 
 #include <QDebug> // qWarning()
+#include <QElapsedTimer>
 #include <QIcon>
 #include <QThreadPool>
 
